cmd/clusterctl/examples/ibmcloud/machines.yaml.template

```diff
@@ -13,7 +13,6 @@ items:
       value:
         apiVersion: "ibmcloudproviderconfig/v1alpha1"
         kind: "IbmcloudMachineProviderSpec"
-        hostname: <host name>
```

Comment:
@xunpan I asked this question and at that time seems we want to use hostname+domain as name, so why we rename this ?

Reply:
we use resource name as the host name. So, this host name is useless. Currently, this is fine for use.

And, I did not delete the internal data structure to make it a feature on demand later.

Reply:
ok, that works for me , just want to make sure I catch up with your changes 
thanks for all the work :)